src/environ/_environ_config.py

```diff
@@ -190,54 +251,9 @@ def to_config(config_cls, environ=os.environ):
 
     This is equivalent to calling ``config_cls.from_environ()``.
     """
-    if config_cls._prefix:
-        app_prefix = (config_cls._prefix,)
-    else:
-        app_prefix = ()
-
-    def default_get(environ, metadata, prefix, name):
-        ce = metadata[CNF_KEY]
-        if ce.name is not None:
-            var = ce.name
-        else:
-            var = ("_".join(app_prefix + prefix + (name,))).upper()
-
-        log.debug("looking for env var '%s'." % (var,))
-        val = environ.get(
-            var,
-            (
-                attr.NOTHING
-                if isinstance(ce.default, attr.Factory)
-                else ce.default
-            ),
-        )
-        if val is RAISE:
-            raise MissingEnvValueError(var)
-        return val
-
-    return _to_config(config_cls, default_get, environ, ())
-
-
-def _to_config(config_cls, default_get, environ, prefix):
-    vals = {}
-    for a in attr.fields(config_cls):
-        try:
-            ce = a.metadata[CNF_KEY]
-        except KeyError:
-            continue
-        if ce.sub_cls is None:
-            get = ce.callback or default_get
-            val = get(environ, a.metadata, prefix, a.name)
-        else:
-            val = _to_config(
-                ce.sub_cls,
-                default_get,
-                environ,
-                prefix + ((a.name if prefix else a.name),),
```

Comment:
I didn't really understand this ternary so I simplified it to just `a.name` in the new location above (cont there)...

Reply:
Yeah the ternary looks very wrong! I think it should be `(prefix + a.name) if prefix else a.name),`? 🤔

Reply:
Sounds about right. I'll change the moved line to effectively match your snippet.